module_text_llm/module_text_llm/helpers/models/openai.py

```diff
@@ -1,254 +1,44 @@
-            available_models[OPENAI_PREFIX + model_name] = OpenAI(
-                model=model_name,
-                openai_api_key=openai_api_key,
-                client="",
-                temperature=0
-            )
-
-    if azure_openai_available:
-        azure_openai_api_key = os.environ["LLM_AZURE_OPENAI_API_KEY"]
-        azure_openai_api_base = os.environ["LLM_AZURE_OPENAI_API_BASE"]
-        azure_openai_api_version = os.environ["LLM_AZURE_OPENAI_API_VERSION"]
-
-        for model_type, Model in [("chat_completion", AzureChatOpenAI), ("completion", AzureOpenAI)]:
-            for deployment_name, deployment in available_deployments[model_type].items():
-                available_models[AZURE_OPENAI_PREFIX + deployment_name] = Model(
-                    model=deployment.model,
-                    deployment_name=deployment_name,
-                    openai_api_base=azure_openai_api_base,
-                    openai_api_version=azure_openai_api_version,
-                    openai_api_key=azure_openai_api_key,
-                    client="",
-                    temperature=0
-                )
-
-    return available_models
-
-
-_model_aliases = {
-    "gpt-35-turbo": "gpt-3.5-turbo",
-}
-
-# Hardcoded because openai can't provide a trustworthly api to get the list of models and capabilities...
-openai_models = {
-    "chat_completion": [
-        "gpt-4",
-        # "gpt-4-32k", # Not publicly available
-        "gpt-3.5-turbo",
-        "gpt-3.5-turbo-16k"
-    ],
-    "completion": [
-        "text-davinci-003",
-        "text-curie-001",
-        "text-babbage-001",
-        "text-ada-001",
-    ],
-    "fine_tuneing": [
-        "davinci",
-        "curie",
-        "babbage",
-        "ada",
-    ]
-}
-available_deployments = _get_available_deployments(openai_models, _model_aliases)
-available_models = _get_available_models(openai_models, available_deployments)
-
+openai_available = bool(os.environ.get("OPENAI_API_KEY"))                                      
+azure_openai_available = bool(os.environ.get("AZURE_OPENAI_API_KEY"))
+
+available_models: Dict[str, BaseLanguageModel] = {}
+
+if openai_available:
+    openai.api_type= "openai"
+    for model in openai.models.list():
+        if model.owned_by == "openai":
+            available_models[OPENAI_PREFIX + model.id] = ChatOpenAI(#type:ignore
+            model=model.id,
+        )
+
+if azure_openai_available:
+    # Will be replaced in the future
+    for deployment in os.environ['AZURE_DEPLOYMENTS'].split(','):
+                available_models[AZURE_OPENAI_PREFIX + deployment] = AzureChatOpenAI(deployment_name=deployment, temperature=0, client="")
+  
 if available_models:
     logger.info("Available openai models: %s", ", ".join(available_models.keys()))
 
     OpenAIModel = Enum('OpenAIModel', {name: name for name in available_models})  # type: ignore
-
-
-    default_model_name = "gpt-3.5-turbo"
+    default_model_name = AZURE_OPENAI_PREFIX+"gpt-35-turbo"
```

Comment:
```suggestion
    default_model_name = "gpt-35-turbo"
```
This is just the default model in the case that we have nothing configured as `LLM_DEFAULT_MODEL` for the non-azure case.